### PR TITLE
Dev 6344 create populate reporting agency missing tas

### PR DESCRIPTION
**Pull request by brett-varney**
**Description:**
Create and populate reporting_agency_missing_tas table, supporting About the Data reports.

**Technical details:**
Copied and adapted from reporting_agency_tas table, just finds and sums GTAS rows missing from appropriation_account_balances.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated--no API update
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Frontend (N/A)
    - [x] Operations (N/A)
    - [x] Domain Expert (N/A)
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-6344](https://federal-spending-transparency.atlassian.net/browse/DEV-6344):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download) (N/A)
    - [x] Before / After data comparison--spot-validated after populating

### CODE REVIEW

File: usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        mommy.make(
+            "references.GTASSF133Balances",
+            treasury_account_identifier=gtas["treasury_account_identifier"],
+            fiscal_year=gtas["fiscal_year"],
+            fiscal_period=gtas["fiscal_period"],
+            obligations_incurred_total_cpe=gtas["obligations_incurred_total_cpe"],
+        )
+
+
+def test_run_script(setup_test_data):
```
Comment (tony-sappe) [inline usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py:97]: Difficult for me to tell if this single test function captures enough edge cases. It looks like a single happy path

**Review by tony-sappe — CHANGES_REQUESTED** [review]
Some requests for changes. Haven't tested anything locally. @alburde1 did you and @brett-varney work together on the SQL?

**Comment by brett-varney** [comment]
> 
> 
> Some requests for changes. Haven't tested anything locally. @alburde1 did you and @brett-varney work together on the SQL?

She helped validate my thinking.

File: usaspending_api/reporting/migrations/0001_initial.py [hunk 2]
```diff
@@ -1,5 +1,3 @@
-# Generated by Django 2.2.17 on 2020-11-19 17:24
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/migrations/0001_initial.py:1]: Does that mean it should be in another file? What are name & location conventions?

File: usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py (hunk 1 above)
Comment (brett-varney) [reply usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py:97]: Should I comment the cases? It includes results for a case that needed to be summed, one that is a single row (still summed, but...), and ones that should not be in the results.

File: usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    """ Test that the populate_reporting_agency_missing_tas script acts as expected """
+    sql_path = str(settings.APP_DIR / "reporting/management/sql/populate_reporting_agency_missing_tas.sql")
+
+    with open(sql_path) as f:
+        test_sql = f.read()
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py]: done

File: usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    with connection.cursor() as cursor:
+        cursor.execute(test_sql)
+        cursor.execute("SELECT * FROM reporting_agency_missing_tas ORDER BY fiscal_year, fiscal_period, toptier_code")
+        results = cursor.fetchall()
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py]: done, thank you

File: usaspending_api/reporting/migrations/0001_initial.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tony-sappe) [reply usaspending_api/reporting/migrations/0001_initial.py:1]: Any change to a Django model should be captured as a migration by running:

```python3 manage.py makemigrations```

Which will create the necessary file(s). This is the expected methodology for tracking DDL changes in Django-managed projects. There are edgecases, but those are edgcases and require a clear justification.

File: usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py:99]: good to know

File: usaspending_api/reporting/migrations/0001_initial.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/migrations/0001_initial.py:1]: I see... re-done

[the remark this answers is not in the document]
Comment (tony-sappe) [reply usaspending_api/reporting/migrations/0001_initial.py:1]: Great. Can you revert the changes to this file?

[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/migrations/0001_initial.py]: done

**Comment by brett-varney** [comment]
@tony-sappe For all the currently unchecked boxes, does this ticket need them?

**Review by tony-sappe — APPROVED** [review]
